crates/cargo-bsp/src/project_model/rust_extension/package.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::{HashMap, HashSet, VecDeque};
+
+fn resolve_origin(package: &mut RustPackage, workspace: &ProjectWorkspace) {
+    if workspace.is_package_part_of_workspace(&package.id) {
+        package.origin = RustPackageOrigin::Workspace;
+    } else {
+        package.origin = RustPackageOrigin::Dependency;
+    }
+}
+
+fn resolve_enabled_features(package: &mut RustPackage, nodes: &[cargo_metadata::Node]) {
+    if let Some(n) = find_node(
+        nodes,
+        &package.id,
+        "Proceeding with empty enabled features.",
+    ) {
+        package.enabled_features = n.features.clone()
+    }
+}
+
+fn metadata_features_to_rust_extension_features(
+    metadata_features: HashMap<String, Vec<String>>,
+) -> Vec<RustFeature> {
+    metadata_features
+        .into_iter()
+        .map(|(f, deps)| RustFeature {
+            name: f,
+            dependencies: deps,
+        })
+        .collect()
+}
+
+fn metadata_package_to_rust_extension_package(
+    metadata_package: cargo_metadata::Package,
+) -> RustPackage {
+    let all_targets = metadata_targets_to_rust_extension_targets(metadata_package.targets);
+    RustPackage {
+        id: metadata_package.id.clone().to_string(),
+        root_url: file_uri(metadata_package.manifest_path.parent().unwrap().to_string()),
+        name: metadata_package.name.clone(),
+        version: metadata_package.version.to_string(),
+        edition: metadata_edition_to_bsp_edition(metadata_package.edition),
+        source: metadata_package.source.map(|s| s.to_string()),
+        features: metadata_features_to_rust_extension_features(metadata_package.features),
+        // In our case targets = all_targets. This field is needed for Bazel
+        targets: all_targets.clone(),
+        all_targets,
+        // The rest of the fields is resolved later
+        ..RustPackage::default()
+    }
+}
+
+/// Returns a list of rust extension packages from which provided targets depend on
+pub fn get_rust_packages_related_to_targets(
+    workspace: &ProjectWorkspace,
+    metadata: &cargo_metadata::Metadata,
+    targets: &[BuildTargetIdentifier],
+) -> Vec<RustPackage> {
+    let target_related_packages_names: Vec<String> = targets
+        .iter()
+        .filter_map(|t| workspace.get_package_related_to_target(t))
+        .flat_map(|p| find_all_packages(p, &metadata.packages))
+        .collect::<HashSet<_>>()
+        .into_iter()
+        .collect();
+
+    let nodes = get_nodes_from_metadata(metadata);
+
+    target_related_packages_names
+        .iter()
+        .map(|n| {
+            let package = metadata
+                .packages
+                .iter()
+                .find(|p| p.name == *n)
+                .unwrap()
+                .clone();
+            let mut rust_package = metadata_package_to_rust_extension_package(package);
+            resolve_origin(&mut rust_package, workspace);
+            resolve_enabled_features(&mut rust_package, &nodes);
```

Comment:
enabledFeatures resolvowane z nodes, które są z metadata wołanego z all features, skutek enabledFeatures = all features. Podjąć decyzję co z tym robimy. Proponuje wysyłać puste i dodać komentarz. W celu lepszej dokumentacji można nawet w readme o tym napisać

Reply:
Czy jeżeli nie będziemy chcieli się tym zajmować to nie lepiej zostawić domyślnie wszystkie features jako enabled zamiast żadnych?

Reply:
wydaje mi się, że to jak jest to zrobione w ten sposób jest baardzo nieoczywiste i to skąd wynika, że bierzemy all features jest przypadkowe, a nie zamierzone, dlatego akurat to bym zmienił

Reply:
jeśli nie chcesz się tym zajmować, to ja mogę na to zerknąć, ale to raczej dopiero w niedzielę

Reply:
Ja chyba nie widzę problemu -> normalnie intellij-rust wywołuje cargo metadata z --all-features i zbiera enabledFeatures dokładnie tak jak my, czyli bierze features z resolveNode. Więc nie wiem czemu mielibyśmy się zachowywać inaczej niż plugin normalnie

Reply:
okej